### PR TITLE
TimeZone Transformer

### PR DESCRIPTION
**Pull request by frederikgjensen**
A transformer that uses longitude and latitude values to extract timezones.

### CODE REVIEW

File: src/atc/transformers/timezone_transformer.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    process(df: DataFrame):
+        returns the input DataFrame with a TimeZone column
+    """
+
+    def __init__(
+        self,
+        longitude_col: float,
+        latitude_col: float,
+        dataset_input_key: str,
+        dataset_output_key: str,
```
Comment (LauJohansson) [inline src/atc/transformers/timezone_transformer.py:36]: dataset_input_key and data_output_key are optional in the extended transformer. Is it on purpurse that they are required here?

File: src/atc/transformers/timezone_transformer.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import pyspark.sql.functions as F
+from pyspark.sql import DataFrame
+from timezonefinder import TimezoneFinder
+
+from atc.etl.extended_transformer import ExtendedTransformer
```
Comment (LauJohansson) [inline src/atc/transformers/timezone_transformer.py:5]: Thinks that it is possible to write: "from atc.etl import ExtendedTransformer"

File: tests/local/transformers/test_timezone_transformer.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+class TimeZoneTransformerTest(DataframeTestCase):
+    def test_timezone_transformer(self):
+        input_schema = T.StructType(
+            [
+                T.StructField("Latitude", T.DoubleType(), True),
+                T.StructField("Longitude", T.DoubleType(), True),
+            ]
+        )
+
+        input_data = [(51.519487, -0.083069), (55.6761, 12.5683)]
```
Comment (LauJohansson) [inline tests/local/transformers/test_timezone_transformer.py:17]: also test the None logic. Where either lat or lon are None.

File: tests/local/transformers/test_timezone_transformer.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        expected_data = [
+            (51.519487, -0.083069, "Europe/London"),
+            (55.6761, 12.5683, "Europe/Copenhagen"),
+        ]
+
+        self.assertDataframeMatches(
+            df=transformed_df,
+            columns=None,
+            expected_data=expected_data,
+        )
```
Comment (LauJohansson) [inline tests/local/transformers/test_timezone_transformer.py:37]: What happens if some of the lat/lon values exeeds the maximum/minium expected values?

File: src/atc/transformers/timezone_transformer.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        )
+        self.longitude_col = longitude_col
+        self.latitude_col = latitude_col
+        self.column_output_name = column_output_name
+
+    def process(self, df: DataFrame) -> DataFrame:
+        """
+        Extracts a timezone using latitude and longitude columns,
+        and outputs the input DataFrame with a TimeZone column.
+        """
```
Comment (LauJohansson) [inline src/atc/transformers/timezone_transformer.py:50]: Maybe throw exeption if the longitude_col or latitude_col is not in the dataframe?

**Review by LauJohansson — CHANGES_REQUESTED** [review]

File: tests/local/transformers/test_timezone_transformer.py (hunk 4 above)
Comment (frederikgjensen) [reply tests/local/transformers/test_timezone_transformer.py:37]: ValueError: The given latitude 120.2 is out of bounds

File: src/atc/transformers/timezone_transformer.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        columns = df.columns
+
+        if self.latitude_col not in columns:
+            raise Exception("The specified latitude column is not in the DataFrame")
```
Comment (LauJohansson) [inline src/atc/transformers/timezone_transformer.py:55]: To further improve your code you could consider adding something like this to the "atc_exceptions.py" file. 

```python
class ColumnDoesNotExistException(AtcException):
    pass
```
Then import it to the file:
```python
from atc.atc_exceptions import ColumnDoesNotExistException
```

And then use it as 
```python
raise ColumnDoesNotExistException("The specified latitude column is not in the DataFrame")
```

This way it becomes easier for the uers to understand the exeptions. We can then reuse the exeption in other classes too.

**Review by LauJohansson — CHANGES_REQUESTED** [review]

**Review by LauJohansson — APPROVED** [review]
